mod_ci/controllers.py

```diff
@@ -866,6 +867,44 @@ def show_maintenance():
     }
 
 
+@mod_ci.route('/blocked_users')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer()
+def blocked_users():
+    return {
+        'blocked_users': BlockedUsers.query.order_by(BlockedUsers.userID)
+    }
+
+
+@mod_ci.route('/blocked_users/add', method=['POST'])
+@login_required
+@check_access_rights([Role.admin])
+def add_user_to_blacklist():
+    form = AddUsersToBlacklist()
+    if form.validate_on_submit():
+        blocked_user = BlockedUsers(form.userID.data, form.comment.data)
+        if blocked_user == BlockedUsers.query.filter_by(userID=form.userID.data).first():
+            flash('User already blocked.')
+            return redirect(url_for('.blocked_users'))
+        g.db.add(blocked_user)
+        g.db.commit()
```

Comment:
You forgot to redirect the user, so the user will see a blank page if it all goes fine.